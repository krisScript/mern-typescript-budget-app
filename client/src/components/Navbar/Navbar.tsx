import React, {
  FunctionComponent,
  useContext,
  useState,
  useEffect,
} from 'react';
import { NavLink } from 'react-router-dom';
import { observer } from 'mobx-react-lite';
import RootStoreContext from '../../stores/RootStore/RootStore';
const Navbar: FunctionComponent = observer(
  (): JSX.Element => {
    const { authStore } = useContext(RootStoreContext);
    console.log(authStore.authState.user.username, authStore.authState.isAuth);
    return (
      <nav className="navbar ">
        <div className="navbar-menu is-active">
          <NavLink to="/add-expense" className="navbar-item">
            Add Expense
          </NavLink>
          <NavLink to="/expenses" className="navbar-item">
            View Expenses
          </NavLink>
          <div className="navbar-end">
            <div className="navbar-item">
              <div className="buttons">
                {authStore.authState.isAuth ? (
                  <button
                    onClick={() => authStore.resetAuthState()}
                    className="button is-primary"
                    data-testid="logout"
                  >
                    Logout
                  </button>
                ) : (
                  <>
                    <NavLink
                      to="/sign-up"
                      className="button is-primary"
                      data-testid="login"
                    >
                      <strong>Sign up</strong>
                    </NavLink>
                    <NavLink
                      to="/login"
                      className="button is-primary"
                      data-testid="signup"
                    >
                      <strong>Login</strong>
                    </NavLink>
                  </>
                )}
              </div>
            </div>
          </div>
        </div>
      </nav>
    );
  },
);

export default Navbar;
