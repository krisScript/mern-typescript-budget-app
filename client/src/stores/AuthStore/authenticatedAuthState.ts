import AuthState from '../../interfaces/AuthState';
const authencticatedAuthState: AuthState = {
  isAuth: true,
  token: 'testToken',
  expiryDate: 'date',
  user: {
    userId: '10',
    username: 'testUser',
    email: 'testEmail@mail.com',
  },
};

export default authencticatedAuthState;
