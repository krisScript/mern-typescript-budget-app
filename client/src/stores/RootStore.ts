import { create } from 'mobx-persist';
import { createContext } from 'react';
import AuthStore from './AuthStore/AuthStore';
export class RootStore {
  public authStore = new AuthStore();
}

const RootStoreContext = createContext(new RootStore());
export default RootStoreContext;
