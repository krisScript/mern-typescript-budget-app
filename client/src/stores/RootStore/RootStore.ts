import { create } from 'mobx-persist';
import { createContext } from 'react';
import AuthStore from '../AuthStore/AuthStore';
import ModalStore from '../ModalStore/ModalStore';
const hydrate = create({
  storage: localStorage,
  jsonify: true,
});
export class RootStore {
  public authStore = new AuthStore();
  public modalStore = new ModalStore();
  public constructor() {
    hydrate('authStore', this.authStore);
  }
}

const RootStoreContext = createContext(new RootStore());
export default RootStoreContext;
