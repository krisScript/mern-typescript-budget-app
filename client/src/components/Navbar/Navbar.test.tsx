import React, { createContext } from 'react';
import { render, cleanup, fireEvent, getByTestId } from 'react-testing-library';
import { BrowserRouter } from 'react-router-dom';
import Navbar from './Navbar';
import { RootStore } from '../../stores/RootStore/RootStore';
import authenticatedAuthState from '../../stores/AuthStore/authenticatedAuthState';
const rootStore = new RootStore();
rootStore.authStore.setAuthState(authenticatedAuthState);
const Context = createContext(rootStore);
// , {
//   wrapper: ({ children }) => (
//     <BrowserRouter>
//       <Context.Provider value={rootStore}>{children}</Context.Provider>
//     </BrowserRouter>
//   ),
// }

describe('<Navbar />', (): void => {
  const { container, rerender, queryByTestId, getByText } = render(<Navbar />, {
    wrapper: ({ children }) => <BrowserRouter>{children}</BrowserRouter>,
  });

  it('snapshot', (): void => {
    expect(container).toMatchSnapshot();
  });
  it('should have Login and SignUp links if isAuth is false', async (): Promise<
    void
  > => {
    const LoginLink = queryByTestId('login');
    const SignUpLink = queryByTestId('signup');
    const LogoutLink = queryByTestId('logout');
    expect(LoginLink).toBeTruthy();
    expect(SignUpLink).toBeTruthy();
    expect(LogoutLink).toBeNull();
  });
  // it('should have logout if isAuth is true', (): void => {
  //   const LoginLink = queryByTestId('login');
  //   const SignUpLink = queryByTestId('signup');
  //   const LogoutLink = queryByTestId('logout');
  //   expect(LoginLink).toBeTruthy();
  //   expect(SignUpLink).toBeTruthy();
  //   expect(LogoutLink).toBeNull();
  // });
});
