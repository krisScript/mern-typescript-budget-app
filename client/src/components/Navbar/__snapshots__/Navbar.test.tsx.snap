// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> snapshot 1`] = `
<div>
  <nav
    class="navbar "
  >
    <div
      class="navbar-menu is-active"
    >
      <a
        class="navbar-item"
        href="/add-expense"
      >
        Add Expense
      </a>
      <a
        class="navbar-item"
        href="/expenses"
      >
        View Expenses
      </a>
      <div
        class="navbar-end"
      >
        <div
          class="navbar-item"
        >
          <div
            class="buttons"
          >
            <a
              class="button is-primary"
              data-testid="login"
              href="/sign-up"
            >
              <strong>
                Sign up
              </strong>
            </a>
            <a
              class="button is-primary"
              data-testid="signup"
              href="/login"
            >
              <strong>
                Login
              </strong>
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
